pkg/agent/baker.go

```diff
@@ -435,6 +435,36 @@ func getContainerServiceFuncMap(config *datamodel.NodeBootstrappingConfiguration
 			}
 			return 0
 		},
+		"ShouldConfigContainerdUlimits": func() bool {
+			return profile.GetCustomLinuxOSConfig().GetUlimitConfig() != nil
```

Comment:
could `GetCustomLinuxOSConfig` ever return nil?

Reply:
Yeah,

```go
func (a *AgentPoolProfile) GetCustomLinuxOSConfig() *CustomLinuxOSConfig {
	if a == nil {
		return nil
	}
	return a.CustomLinuxOSConfig
}
```
but if it does then the whole chain will simply return nil, its written in the protobuf getter style.